client/src/components/Place.js

```diff
@@ -1,174 +1,270 @@
-        <Col sm={2}><Form.Control type="text" placeholder="To" defaultValue={p.opening_hours.wed_close} /></Col>
-      </Form.Group>
-
-      <Form.Group as={Form.Row} controlId="formHorizontalOpeningHours5">
-        <Form.Label column sm={1}>Thu</Form.Label>
-        <Col sm={2}><Form.Control type="text" placeholder="From" defaultValue={p.opening_hours.thu_open} /></Col>-
-        <Col sm={2}><Form.Control type="text" placeholder="To" defaultValue={p.opening_hours.thu_close} /></Col>
-      </Form.Group>
-
-      <Form.Group as={Form.Row} controlId="formHorizontalOpeningHours6">
-        <Form.Label column sm={1}>Fri</Form.Label>
-        <Col sm={2}><Form.Control type="text" placeholder="From" defaultValue={p.opening_hours.fri_open} /></Col>-
-        <Col sm={2}><Form.Control type="text" placeholder="To" defaultValue={p.opening_hours.fri_close} /></Col>
-      </Form.Group>
-
-      <Form.Group as={Form.Row} controlId="formHorizontalOpeningHours7">
-        <Form.Label column sm={1}>Sat</Form.Label>
-        <Col sm={2}><Form.Control type="text" placeholder="From" defaultValue={p.opening_hours.sat_open} /></Col>-
-        <Col sm={2}><Form.Control type="text" placeholder="To" defaultValue={p.opening_hours.sat_close} /></Col>
-      </Form.Group>
-    </Form>
-  );
-};
+  defaultState = {
+    id: null,
+    name: null,
+    description: null,
+    address: null,
+    website: null,
+    phone: null,
+    email: null,
+    lat: null,
+    lon: null,
+    opening_hours: {
+      sun_open: null,
+      sun_close: null,
+      mon_open: null,
+      mon_close: null,
+      tue_open: null,
+      tue_close: null,
+      wed_open: null,
+      wed_close: null,
+      thu_open: null,
+      thu_close: null,
+      fri_open: null,
+      fri_close: null,
+      sat_open: null,
+      sat_close: null,
+    },
+    tags: [],
+  };
+
+  updateState = (e: SyntheticInputEvent<*>) => {
+    this.setState({ [e.target.id]: e.target.value });
+  }
+
+  updateTags = (e: SyntheticInputEvent<*>) => {
+    this.setState({ tags: e.target.value.split(',') });
+  }
+
+  updateTime = (e: SyntheticInputEvent<*>) => {
+    let v = e.target.value;
+    if (isNaN(Number(v))) {
+      return;
+    }
+    switch (v.length) {
+    case 1:
+      v = '0' + v + ':00';
+      break;
+    case 2:
+      v = v + ':00';
+      break;
+    case 3:
+      v = v.charAt(0) + ':' + v.slice(1);
+      break;
+    case 4:
+      v = v.slice(0, 2) + ':' + v.slice(2);
+      break;
+    default:
+      v = null;
+    }
+    e.target.value = v;
+    let parts = e.target.id.split('.');
+    this.setState(prevState => ({
+      opening_hours: {
+        ...prevState.opening_hours,
+        [parts[1]]: v,
+      },
+    }));
+  }
+
+  handleSave = () => {
+    this.props.doSavePlace(this.state);
+  }
+
+  render() {
+    const { place, reviews, tags } = this.props;
+    let p = place || {};
```

Comment:
Not sure why is this needed. The screen that renders this component can render it only if the place exists (loaded and everything). Same for the default state above and the opening hours below.